### DISCUSSION TITLE
Add datasource for AWS RDS Performance Insights metrics

### DISCUSSION THREAD

**Discussion by lobeck** [grafana/grafana#38326] [Data Sources]
**What would you like to be added**:

AWS Performance Insights offer some buildin Metrics, that are not available through CloudWatch. They are a great addition to the existing CloudWatch metrics and provide much more insights into the database behaviour.

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights_Counters.html#USER_PerfInsights_Counters.PostgreSQL

**Why is this needed**:

Visibility into RDS is difficult, as the CloudWatch metrics are only very abstract and don't yield much information on their own. With the additional metrics from Performance Insights, one could get a much better few into the database and help with troubleshooting

**Comment by aocenas** [comment]
@lobeck thanks for the request. This is not on our roadmap at the moment so if you didn't already you could search whether there isn't existing 3rd party plugin for this as not all plugins are listed on https://grafana.com/grafana/plugins.

**Comment by Xopherus** [comment]
Would it be possible to add the metrics which are published by PI through Cloudwatch? There's [DBLoad, DBLoadCPU, and DBLoadNonCPU](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.Cloudwatch.html). I don't see these available, though I am on Grafana 6.7 (forgive me if they were added in Grafana 7).

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by mrsemmix** [comment]
Anything happening with this ?

**Comment by sysadmin1139** [comment]
We've found that if you *hand type* in these metrics, they will render. The inspect of our graph looks like:

```
REMOVE_EMPTY(SEARCH('{"AWS/RDS","DBInstanceIdentifier"} MetricName="DBLoad"', 'Average', 60))
```
![DBLoad-in-Dashboard](https://user-images.githubusercontent.com/812860/162487135-c84a525c-fea3-419c-9b04-d47a0bd45b8c.png)
This value does not show up in the dropdown. To get it:

1. Type in `DBLoad`
2. Above the text field you will see `Create: DBLoad` Hit Enter to create it

This should let you render the metric.

**Reply by lorelei-rupp-imprivata** [reply]
Have you found a way to slice the DBLoad by the dimensions that you see on the performance insights dashboard? That is what I am interested in

**Reply by mundabra** [reply]
You can do that using RDS Performance Insights API

**Comment by lorelei-rupp-imprivata** [comment]
We would really like to have similar graphs in Grafana for what we see in performance insights, as going to two different places for dashboards is not what we want. Has anyone found other ways to do this with Grafana?

**Reply by kayketeixeira** [reply]
> We would really like to have similar graphs in Grafana for what we see in performance insights, as going to two different places for dashboards is not what we want. Has anyone found other ways to do this with Grafana?

Do you have any update? We also need this feature

**Reply by mundabra** [reply]
yes @lorelei-rupp-imprivata  @kayketeixeira Please check out this blog and sample code to do that : 

https://aws.amazon.com/blogs/mt/monitoring-amazon-rds-and-amazon-aurora-using-amazon-managed-grafana/

**Comment by hassanejaz** [comment]
This would be so useful!

**Comment by marekaf** [comment]
One way is to use Lambda to pull metrics from PI API and push it to CloudWatch https://aws.amazon.com/blogs/database/importing-amazon-rds-performance-insights-counter-metrics-to-amazon-cloudwatch/

PI Data Source would be a better option. 

AWS fixing this and making PI metrics available in CloudWatch would be the best option for everybody imho.

**Comment by craig-morton** [comment]
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.Cloudwatch.html

**Reply by marekaf** [reply]
these are only 3 metrics

**Reply by mundabra** [reply]
@marekaf  Are you looking for any specific metrics?

**Reply by marekaf** [reply]
I was specifically looking for OOMkills, I will check out what's in CW now, thanks a lot!

**Comment by mundabra** [comment]
We have published a new AWS blog which addresses this use case, leverage RDS Performance Insights API to publish Performance Insight metrics to CloudWatch and also provide Grafana dashboard to view these metrics in Amazon Managed Grafana. You can also view these metrics in CloudWatch dashboards.

https://aws.amazon.com/blogs/mt/monitoring-amazon-rds-and-amazon-aurora-using-amazon-managed-grafana/

Please provide your feedback if you want additional functionality related to your specific use case.

**Reply by lorelei-rupp-imprivata** [reply]
You still have to deploy a lambda though with this right from what im reading? thats what we dont want to do, why not have a datasource that can pull this stuff directly from performance insights

**Comment by sarahzinger** [comment]
Hi Folks, we've created an issue for this feature request for us to consider when choosing future plugins to build. Thanks so much for the suggestion! If you have other details you'd like to add feel free to add them here: https://github.com/grafana/grafana/issues/78139

Until we can prioritize this work though, definitely checkout the work around suggested up above, and if you're feeling ambitious and want to [build your own plugin](https://grafana.com/developers/plugin-tools/tutorials/build-a-data-source-plugin) we also support that! 

Thanks again for the suggestion!
